src/pystage/core/gui.py

```diff
@@ -183,11 +186,31 @@ def __init__(self, sprite):
 
     def say(self, text: str, border=Bubble.SAY):
         # print(text)
+        if isinstance(text, (int, float)):
+            text = str(round(text, 2) if text % 1 > 0 else int(text))
```

Comment:
@Zhonglin-Niu : what does `text % 1 > 0` mean? This is very unintuitive code and should have a comment, or be rewritten so the average person can understand it.

Reply:
> @Zhonglin-Niu : what does `text % 1 > 0` mean? This is very unintuitive code and should have a comment, or be rewritten so the average person can understand it.

Means whether `text` has a non-zero fractional part.

Reply:
OK: a comment should be in the code above that line then, IMO.